includes/user-functions.php

```diff
@@ -270,9 +270,19 @@ function give_count_donations_of_donor( $user = null ) {
  */
 function give_donation_total_of_user( $user = null ) {
 
-	$stats = give_get_donation_stats_by_user( $user );
+    // Logged in?
```

Comment:
@mcdonagg thank you for pr.

As I can see in this pr, you set current logged in user as the fallback to get the donation. We do not want to assume anything that why is asking for a user id to return the result.

Instead of finding user id in core function, I will suggest to create a logic outside of this function and then pass user id to this function to get the result.

Reply:
Sounds good, it was just a small suggestion to make this function more in line with similar functions you have that do assume the logged in user if not provided. I did not see any negative ramifications of doing so, but it does not matter to me. Thank you for looking at the request.